### ISSUE TITLE
want a standalone iterator for ranges under NUMA

### ISSUE THREAD

**Issue by vasslitvinov** [chapel-lang/chapel#12329]
The standalone iterator for ranges that we currently have does not apply to NUMA locale model.

We would like to add one that works under NUMA.

See, for example, #12322, which was due to the standalone iterators failing to compile under NUMA, due to their reliance on the standalone iterator for ranges, in the following files:
```
test/parallel/forall/may-vs-must-par/iters.chpl
test/parallel/forall/may-vs-must-par/datas.chpl
```

**Comment by vasslitvinov** [comment]
BTW the absence of the said range iterator has given us a good idea of how well (or, rather, not well) the iterator forwarding idiom is supported in Chapel.

**Comment by cassella** [comment]
With the NUMA locale model deprecated, should this issue be closed?

**Comment by vasslitvinov** [comment]
That's right. Closing, as we do not provide the NUMA locale model any longer.
